aws/resource_aws_appautoscaling_scheduled_action.go

```diff
@@ -0,0 +1,222 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/applicationautoscaling"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+const awsAppautoscalingScheduleTimeLayout = "2006-01-02T15:04:05Z"
+
+func resourceAwsAppautoscalingScheduledAction() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAppautoscalingScheduledActionCreate,
+		Read:   resourceAwsAppautoscalingScheduledActionRead,
+		Delete: resourceAwsAppautoscalingScheduledActionDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"service_namespace": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					validServiceNamespaces := []string{"ecs", "elasticmapreduce", "ec2", "appstream", "dynamodb"}
+					for _, str := range validServiceNamespaces {
+						if value == str {
+							return
+						}
+					}
+					errors = append(errors, fmt.Errorf("expected %s to be one of %v, got %s", k, validServiceNamespaces, value))
+					return
+				},
+			},
+			"resource_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"scalable_dimension": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					validDimensions := []string{"ecs:service:DesiredCount", "ec2:spot-fleet-request:TargetCapacity",
+						"elasticmapreduce:instancegroup:InstanceCount", "appstream:fleet:DesiredCapacity",
+						"dynamodb:table:ReadCapacityUnits", "dynamodb:table:WriteCapacityUnits",
+						"dynamodb:index:ReadCapacityUnits", "dynamodb:index:WriteCapacityUnits"}
```

Comment:
While offline, plan-time validation is usually useful I'm afraid it could be counter-productive here as the list of valid dimensions may and will expand and it did recently. I think we'd play a constant catch-up with AWS here and rushing to cut a release just to support a new dimension just because of this strict validation.
Theoretically we could validate for something like `[a-zA-Z]+:[a-zA-Z]+:[a-zA-Z]+` but frankly I'd just rather avoid the validation altogether here.

Reply:
[docs](https://docs.aws.amazon.com/ApplicationAutoScaling/latest/APIReference/API_PutScheduledAction.html#ApplicationAutoScaling-PutScheduledAction-request-ScalableDimension) lists constant strings so I removed `ValidateFunc`.